data/data/agent/files/usr/local/bin/set-hostname.sh

```diff
@@ -35,7 +36,29 @@ do
         echo "Host has matching MAC address: ${filename}" 1>&2
         echo "Setting hostname to ${HOSTNAME}" 1>&2
         hostnamectl set-hostname "${HOSTNAME}"
+	hostname_set=1
     else
         echo "MAC address, ${filename}, does not exist on this host" 1>&2
     fi
 done
+
+if [[ $hostname_set -eq 0 ]]; then
+   ips=$(hostname -I)
+   while [[ $ips == "" ]]
+   do
+      echo "Waiting for IPs to appear for reverse DNS" 1>&2
+      sleep 5
+      ips=$(hostname -I)
+   done
+
+   for ip in ${ips}
+   do
+      echo "Reverse dns for $ip" 1>&2
+      hostname=$(dig -x $ip +short)
+      # If hostname returned, configure it
+      if [[ $hostname != "" ]]; then
+         hostnamectl set-hostname "${hostname}"
```

Comment:
what if multiple names are returned?

Reply:
Changed to handle multiple names and use the first one.